.github/scripts/generate_matrix.py

```diff
@@ -13,7 +14,15 @@
 suites += [repl_test.replace('dirsrvtests/tests/suites/', '') for repl_test in repl_tests]
 suites.sort()
 
-suites_list = [{ "suite": suite} for suite in suites]
-matrix = {"include": suites_list}
+variants = ['gcc', 'gcc-asan']
```

Comment:
gcc-asan doesn't work with rust, so we can't use this. We need clang-asan to be consistent, especially so to match the correct llvm versions.

Reply:
Ok, I will replace `gcc-asan` with `clang-asan`.

Reply:
@Firstyear, clang with `-fsanitize=address` fails to compile: https://github.com/vashirov/389-ds-base/runs/3554323865#step:6:3541
Any ideas?

Reply:
I found that I filed an issue for this some time ago https://github.com/389ds/389-ds-base/issues/4596
I feel that `clang-asan` would require more efforts than I expected, so I'd like to to keep `gcc-asan` for now and add `clang-asan` in a follow up PR.

Reply:
Didn't we change how lto was managed here to make it work? I'm building and using clang-asan just fine? 

The issue with gcc-asan is that I've noticed more and more that gcc asan and rust's llvm backed asan aren't compatible and can not be mixed, which has led to failures to build. That's why I've been using clang now so that they both have the same llvm backend and asan versions.

Reply:
No, we didn't :( I couldn't make LTO builds with clang and then I switched to other things, forgetting about the issue.
I need some help from you regarding Rust + LTO, I will put the details in the related issue.

I haven't encountered any failures myself. Ideally I'd like to have the full matrix (`gcc`, `gcc-asan`, `clang`, `clang-asan`), so that we can catch these build issues too. But I'm hitting GH actions limit of 256 jobs in the matrix. I have some ideas how to circumvent this with composable actions, will try it next week.

Reply:
Okay, I'll look at the LTO issue later, but I think we can't do gcc-asan + rust at all. I've had so many conflicts between the two ASAN apis as they exist, and a bunch of false positives at this point, that I don't think it should be considered reliable.